src/backend/dx12/src/command.rs

```diff
@@ -1213,17 +1244,128 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
+        filter: image::Filter,
+        regions: T,
     ) where
         T: IntoIterator,
         T::Item: Borrow<com::ImageBlit>
     {
-        unimplemented!()
+        // TODO: only supporting 2D images
+
+        let rtv_pool = {
+            let desc = d3d12::D3D12_DESCRIPTOR_HEAP_DESC {
+                Type: d3d12::D3D12_DESCRIPTOR_HEAP_TYPE_RTV,
+                NumDescriptors: 1,
+                Flags: d3d12::D3D12_DESCRIPTOR_HEAP_FLAG_NONE,
+                NodeMask: 0,
+            };
+
+            let mut heap: *mut d3d12::ID3D12DescriptorHeap = ptr::null_mut();
+            unsafe {
+                (*self.shared.device).CreateDescriptorHeap(
+                    &desc,
+                    &d3d12::IID_ID3D12DescriptorHeap,
+                    &mut heap as *mut *mut _ as *mut *mut _,
+                );
+            };
+            unsafe { ComPtr::from_raw(heap) }
+        };
+        let rtv = unsafe { rtv_pool.GetCPUDescriptorHandleForHeapStart() };
+
+        let filter = match filter {
+            image::Filter::Nearest => d3d12::D3D12_FILTER_MIN_MAG_MIP_POINT,
+            image::Filter::Linear => d3d12::D3D12_FILTER_MIN_MAG_LINEAR_MIP_POINT,
+        };
+
+        for region in regions {
+            let r = region.borrow();
+
+            let first_layer = r.dst_subresource.layers.start;
+            let num_layers = r.dst_subresource.layers.end - first_layer;
+
+            let blit = match r.dst_subresource.aspects {
+                format::Aspects::COLOR => {
+                    // Create RTV of the dst image for the miplevel of the current region
+                    let mut desc = d3d12::D3D12_RENDER_TARGET_VIEW_DESC {
+                        Format: dst.dxgi_format,
+                        ViewDimension: d3d12::D3D12_RTV_DIMENSION_TEXTURE2DARRAY,
+                        u: unsafe { mem::zeroed() },
+                    };
+
+                    *unsafe{ desc.u.Texture2DArray_mut() } = d3d12::D3D12_TEX2D_ARRAY_RTV {
+                        MipSlice: r.dst_subresource.level as _,
+                        FirstArraySlice: first_layer as _,
+                        ArraySize: num_layers as _,
+                        PlaneSlice: 0, //TODO
+                    };
+
+                    unsafe {
+                        (*self.shared.device).CreateRenderTargetView(dst.resource, &desc, rtv);
+                        self.raw.OMSetRenderTargets(
+                            1,
+                            &rtv,
+                            FALSE,
+                            ptr::null(),
+                        );
+                    }
+
+                    self.shared.service_pipes.get_blit_2d_color(dst.dxgi_format, filter)
+                },
+                _ => unimplemented!(),
+            };
+
+            self.set_internal_graphics_pipeline();
+
+            // TODO: support flipping?
+            let viewport = d3d12::D3D12_VIEWPORT {
+                TopLeftX: r.dst_bounds.start.x as _,
```

Comment:
I would prefer us issuing real rectangles for blits (like we do in Metal), since it requires less state switching and can batch multiple blits together.
If you do want to use the viewport, then I don't think your shader even needs to know the destination bounds, it can write to -1 to 1 instead, and those will be scaled properly by the viewport you set here.